### PR TITLE
[GH-13486] Migrate 'components/admin_console/manage_tokens_modal' module and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by tweichart**
#### Summary
This PR migrates the `components/admin_console/manage_tokens_modal` to TypeScript.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/13486

### CODE REVIEW

**Comment by devinbinnie** [comment]
@tweichart Is this ready for review? Looks like all tests are passing.

**Comment by tweichart** [comment]
> @tweichart Is this ready for review? Looks like all tests are passing.

wasn't too sure about the patch for the mattermos-redux type there, but will be checked during review i guess ;-)

**Comment by tweichart** [comment]
> @tweichart Is this ready for review? Looks like all tests are passing.

wasn't too sure about the patch for the mattermos-redux type there, but will be checked during review i guess ;-)

File: components/admin_console/manage_tokens_modal/manage_tokens_modal.tsx [hunk 1]
```diff
@@ -14,58 +16,41 @@ import Avatar from 'components/widgets/users/avatar';
 
 import FormattedMarkdownMessage from 'components/formatted_markdown_message.jsx';
 
-export default class ManageTokensModal extends React.PureComponent {
-    static propTypes = {
+type Props = {
```
Comment (reflog) [inline components/admin_console/manage_tokens_modal/manage_tokens_modal.tsx:19]: Please add the removed comments for the props

**Review by reflog — CHANGES_REQUESTED** [review]
looks great, just one small nit

File: components/admin_console/manage_tokens_modal/manage_tokens_modal.tsx (hunk 1 above)
Comment (tweichart) [reply components/admin_console/manage_tokens_modal/manage_tokens_modal.tsx:19]: thanks for the hint, forgot to add them back! fixed now ;-)

**Comment by tweichart** [comment]
fyi mattermost-redux has just been patched: https://github.com/mattermost/mattermost-redux/commit/a21d5964805911899674c017ae268fb5bf0d1d8c
updated mattermost-redux therefore (2 commits), all seems well now without the patch of types there.

**Review by reflog — APPROVED** [review]
Looks great, thanks!

File: components/admin_console/manage_tokens_modal/manage_tokens_modal.tsx [hunk 2]
```diff
@@ -14,58 +16,63 @@ import Avatar from 'components/widgets/users/avatar';
...
 
-    handleError = (error) => {
+    private handleError = (error: string): void => {
         this.setState({
             error,
         });
     }
 
-    renderContents = () => {
+    public renderContents = (): JSX.Element => {
```
Comment (jespino) [inline components/admin_console/manage_tokens_modal/manage_tokens_modal.tsx:75]: Is there any reason to make it public? I think is not needed.

File: components/admin_console/manage_tokens_modal/manage_tokens_modal.tsx [hunk 3]
```diff
@@ -14,58 +16,63 @@ import Avatar from 'components/widgets/users/avatar';
...
+}
 
-    constructor(props) {
+export default class ManageTokensModal extends React.PureComponent<Props, State> {
+    public constructor(props: Props) {
         super(props);
-        this.state = {error: null};
+        this.state = {
+            error: null,
+        } as State;
```
Comment (jespino) [inline components/admin_console/manage_tokens_modal/manage_tokens_modal.tsx:58]: I think this explicit `as State` is not needed here.

File: components/admin_console/manage_tokens_modal/index.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import ManageTokensModal from './manage_tokens_modal';
+
+type Props = {
+    user?: UserProfile;
+}
+
+function mapStateToProps(state: GlobalState, ownProps: Props) {
+    const userId = ownProps.user ? ownProps.user.id : '';
+
+    const userAccessTokens = (state.entities.admin as AdminState).userAccessTokensByUser;
```
Comment (jespino) [inline components/admin_console/manage_tokens_modal/index.ts:21]: Is this type assertion (`as AdminState`) needed? I don't think so.

**Review by jespino — CHANGES_REQUESTED** [review]
A couple of minor comments, but otherwise looks good to me. Thanks @tweichart! :tada:

File: components/admin_console/manage_tokens_modal/manage_tokens_modal.tsx [hunk 5]
```diff
@@ -14,58 +16,63 @@ import Avatar from 'components/widgets/users/avatar';
...
 
-    handleError = (error) => {
+    private handleError = (error: string): void => {
         this.setState({
             error,
         });
     }
 
-    renderContents = () => {
+    renderContents = (): JSX.Element => {
```
Comment (jespino) [inline components/admin_console/manage_tokens_modal/manage_tokens_modal.tsx:75]: can we say explicitly `private` here?

**Review by jespino — APPROVED** [review]
Now looks good to me. Thanks @tweichart! great work :) :sparkles:

**Comment by tweichart** [comment]
type-checks in pipeline fail due to latest master branch merge (were just fine before), should this be fixed here or can this be ignored?

**Review by ogi-m — COMMENTED** [review]
Hi @tweichart, I'm unable to create a personal access token on the test server:
1. Go to Account Settings > Security
2. Click on Personal Access Tokens > Create Token
3. Enter token description and click "Save"
4. Confirm by clicking "Yes, create" in the popup

**Observed**: Nothing happens
**Expected**: Token details show up

I'm only seeing this issue on the PR, it's working as expected on master.

**Comment by tweichart** [comment]
> Hi @tweichart, I'm unable to create a personal access token on the test server:
> 
> 1. Go to Account Settings > Security
> 2. Click on Personal Access Tokens > Create Token
> 3. Enter token description and click "Save"
> 4. Confirm by clicking "Yes, create" in the popup
> 
> **Observed**: Nothing happens
> **Expected**: Token details show up
> 
> I'm only seeing this issue on the PR, it's working as expected on master.

@ogi-m thanks for the feedback, I've checked and the state of master branch my PR is based on is also not working. actually this error occurs due to this commit here: https://github.com/mattermost/mattermost-webapp/pull/6648 . When i manually roll that back in my PR you can add security tokens again, so i'd say it has nothing to do with my PR.

**Review by ogi-m — APPROVED** [review]
Thanks @tweichart, confirmed I can repro on master. I've open [a ticket](https://mattermost.atlassian.net/browse/MM-30099) for the issue.

Approving the PR 👍

**Pull request merged**